Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py

```diff
@@ -0,0 +1,747 @@
+            params=params
+        )
+
+
+""" HELPER FUNCTIONS """
+
+
+def check_int(arg: Any, arg_name: str, min_val: int = None, max_val: int = None, required: bool = False) -> Optional[int]:
+    """Converts a string argument to a Python int
+    This function is used to quickly validate an argument provided and convert
+    it into an ``int`` type. It will throw a ValueError if the input is invalid
+    or outside the optional range. If the input is None, it will throw a ValueError
+    if required is ``True``, or ``None`` if required is ``False.
+    """
+
+    # check if argument is mandatory
+    if arg is None:
+        if required is True:
+            raise ValueError(f'Missing argument "{arg_name}"')
+        return None
+
+    i: Optional[int] = None
+
+    if isinstance(arg, str):
+        if not arg.isdigit():
+            raise ValueError(f'Integer invalid: "{arg_name}"="{arg}"')
+        try:
+            i = int(arg)
+        except ValueError:
+            raise ValueError(f'Integer invalid: "{arg_name}"="{arg}"')
+    elif isinstance(arg, int):
+        i = arg
+    else:
+        raise ValueError(f'Invalid number: "{arg_name}"')
+
+    # range check
+    if min_val and i < min_val:
+        raise ValueError(f'Integer outside minimum range: "{arg_name}"="{arg}" ("min={min_val}")')
+    if max_val and i > max_val:
+        raise ValueError(f'Integer outside maximum range: "{arg_name}"="{arg}" ("max={max_val}")')
+
+    return i
+
+
+def validate_max_indicators(max_indicators_param: Optional[str]) -> Optional[int]:
+    try:
+        max_indicators = check_int(
+            max_indicators_param,
+            'maxIndicators',
+            None, None, False
+        )
+    except ValueError:
+        return None
+
+    if max_indicators is None:
+        max_indicators = DEFAULT_FETCH_MAX_INDICATORS
+
+    return max_indicators
+
+
+def validate_min_last_observed(min_last_observed_param: Optional[str]) -> Optional[str]:
+    try:
+        min_last_observed = check_int(
+            min_last_observed_param,
+            'minLastObserved',
+            None, None, False
+        )
+    except ValueError:
+        return None
+
+    if min_last_observed is None:
+        min_last_observed = DEFAULT_FETCH_MIN_LAST_OBSERVED
+
+    start_time = datetime.utcnow() - timedelta(days=min_last_observed)
+
+    return start_time.strftime("%Y-%m-%d")
+
+
+def safe_b64_to_hex(i: str) -> Optional[str]:
+    if not i:
+        return None
+
+    try:
+        return base64.urlsafe_b64decode(i).hex()
+    except Exception:
+        return None
+
+
+""" INDICATORS HANDLING FUNCTIONS """
+
+
+def ip_to_demisto_indicator(ip_indicator: Dict[str, Any]) -> Optional[Dict[str, Any]]:
+    value = ip_indicator.get('ip', None)
+    if value is None:
+        return None
+
+    provider_name: Optional[str] = None
+    provider = ip_indicator.get('provider', None)
+    if provider is not None:
+        provider_name = provider.get('name', None)
```

Comment:
suggestion - you can do this instead:
```suggestion
    provider = ip_indicator.get('provider', {})
    provider_name = provider.get('name', None)
```
*same for other occurances*

Reply:
Applied on all the occurences